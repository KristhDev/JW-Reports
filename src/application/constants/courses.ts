export const coursesMessages = {
    EXPORTED_SUCCESS: 'Has exportado tus cursos correctamente. El archivo se encuentra en su carpeta de descargas.',
    PERSON_NAME_REQUIRED: 'El nombre de la persona es requerido.',
    PERSON_NAME_MIN_LENGTH: 'El nombre de la persona debe tener al menos 2 caracteres.',
    PERSON_ABOUT_REQUIRED: 'La información de la persona es requerida.',
    PERSON_ABOUT_MIN_LENGTH: 'La información de la persona debe tener al menos 10 caracteres.',
    PERSON_ADDRESS_REQUIRED: 'La dirección es requerida.',
    PERSON_ADDRESS_MIN_LENGTH: 'La dirección debe tener al menos 10 caracteres.',
    PUBLICATION_REQUIRED: 'La publicación de estudio no puede estar vacía.',
    PUBLICATION_MIN_LENGTH: 'La publicación de estudio debe tener al menos 5 caracteres.',
    UNSELECTED: 'No hay un curso seleccionado.',
    UNSELECTED_FINISH_OR_START: 'No pudes terminar o comenzar de nuevo un curso suspendido.',
    UNSELECTED_DELETE: 'No hay un curso seleccionado para eliminar.',
    UNSELECTED_UPDATE: 'No hay un curso seleccionado para actualizar.',
    FINISHED: 'No puedes suspender o renovar un curso terminado.',
    SUSPENDED_SUCCESS: 'Has suspendido el curso correctamente.',
    RENEW_SUCCESS: 'Has renovado el curso correctamente.',
    DELETED_SUCCESS: 'Has eliminado el curso correctamente.',
    FINISHED_SUCCESS: 'Has terminado el curso correctamente.',
    RESTARTED_SUCCESS: 'Has comenzado de nuevo el curso correctamente.',
    ADDED_SUCCESS: 'Has agregado el curso correctamente.',
    UPDATED_SUCCESS: 'Has actualizado el curso correctamente.',
}