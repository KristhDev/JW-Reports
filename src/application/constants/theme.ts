import { ItemOption } from '../../infrasturcture/interfaces';

/**
 * The theme options to select
 */
export const THEME_OPTIONS: ItemOption[] = [
    { label: 'Modo predeterminado', value: 'default' },
    { label: 'Modo claro', value: 'light' },
    { label: 'Modo oscuro', value: 'dark' }
];