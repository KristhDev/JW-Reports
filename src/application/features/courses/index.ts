export {
    default as coursesReducer,
    INIT_COURSE,
    COURSES_INITIAL_STATE,
    addCourse,
    addCourses,
    addLastLessonInCourse,
    clearCourses,
    removeCourse,
    removeCourses,
    replaceLastLessonInCourse,
    setCourseFilter,
    setCourses,
    setCoursesPagination,
    setCoursesScreenHistory,
    setHasMoreCourses,
    setIsCourseDeleting,
    setIsCourseLoading,
    setIsCoursesExporting,
    setIsCoursesLoading,
    setRefreshCourses,
    setSelectedCourse,
    updateCourse,
    updateLastLessonInCourse
} from './slice';

export * from './types';