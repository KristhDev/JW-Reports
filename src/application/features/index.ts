export * from './auth';
export * from './courses';
export * from './lessons';
export * from './permissions';
export * from './preaching';
export * from './revisits';
export * from './status';
export * from './types';
export * from './ui';