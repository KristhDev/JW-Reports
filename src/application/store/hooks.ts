import { useDispatch, useSelector } from 'react-redux';

/* Store */
import { AppDispatch, RootState } from './store';

/* Defined types hooks */
export const useAppDispatch = useDispatch.withTypes<AppDispatch>();
export const useAppSelector = useSelector.withTypes<RootState>();