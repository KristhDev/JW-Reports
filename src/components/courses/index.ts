export { CourseCard } from './CourseCard';
export { CoursesList } from './CoursesList';