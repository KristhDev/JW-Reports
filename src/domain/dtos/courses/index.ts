export * from './active-suspend.dto';
export * from './create.dto';
export * from './finish-start.dto';
export * from './update.dto';