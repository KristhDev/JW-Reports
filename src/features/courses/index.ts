export {
    default as coursesReducer,
    INIT_COURSE,
    INIT_LESSON,
    addCourse,
    addCourses,
    addLesson,
    addLessons,
    clearCourses,
    removeCourse,
    removeCourses,
    removeLesson,
    removeLessons,
    setCourseFilter,
    setCourses,
    setCoursesPagination,
    setCoursesScreenHistory,
    setHasMoreCourses,
    setHasMoreLessons,
    setIsCourseDeleting,
    setIsCourseLoading,
    setIsCoursesLoading,
    setIsLessonDeleting,
    setIsLessonLoading,
    setIsLessonsLoading,
    setLessons,
    setLessonsPagination,
    setRefreshCourses,
    setRefreshLessons,
    setSelectedCourse,
    setSelectedLesson,
    updateCourse,
    updateLesson
} from './slice';