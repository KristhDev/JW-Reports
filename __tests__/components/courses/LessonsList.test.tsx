import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react-native';
import { MenuProvider } from 'react-native-popup-menu';

/* Components */
import { LessonsList } from '../../../src/components/courses';

/* Features */
import { courseSelectedState, lessonsState } from '../../features/courses';

/* Hooks */
import { useCourses, useTheme } from '../../../src/hooks';

/* Theme */
import { darkColors } from '../../../src/theme';

const deleteLessonMock = jest.fn();
const loadLessonsMock = jest.fn();
const removeLessonsMock = jest.fn();
const setLessonsPaginationMock = jest.fn();
const setSelectedLessonMock = jest.fn();

const emptyMessageTest = 'No haz agregado clases a este curso.';
const titleTest = `Clases del curso con ${ lessonsState.selectedCourse.person_name }`;

/* Mock hooks */
jest.mock('../../../src/hooks/useCourses.ts');
jest.mock('../../../src/hooks/useTheme.ts');

describe('Test in <LessonsList /> component', () => {
    (useCourses as jest.Mock).mockReturnValue({
        state: lessonsState,
        deleteLesson: deleteLessonMock,
        loadLessons: loadLessonsMock,
        removeLessons: removeLessonsMock,
        setLessonsPagination: setLessonsPaginationMock,
        setSelectedLesson: setSelectedLessonMock,
        finishOrStartLesson: jest.fn(),
    });

    (useTheme as jest.Mock).mockReturnValue({
        state: { colors: darkColors },
        BUTTON_TRANSPARENT_COLOR: 'rgba(255, 255, 255, 0.15)'
    });

    beforeEach(() => {
        render(
            <MenuProvider>
                <LessonsList />
            </MenuProvider>
        );

        jest.clearAllMocks();
    });

    it('should to match snapshot', () => {
        expect(screen.toJSON()).toMatchSnapshot();
    });

    it('should render respective props', () => {
        const { getByTestId } = render(
            <MenuProvider>
                <LessonsList />
            </MenuProvider>
        );

        /* Get title of list and check if is text pass for props */
        const titleText = getByTestId('title-text');
        expect(titleText.props.children).toBe(titleTest);
    });

    it('should render message when lessons is empty', () => {

        /* Mock data of useCourses */
        (useCourses as jest.Mock).mockReturnValue({
            state: courseSelectedState,
            deleteLesson: deleteLessonMock,
            loadLessons: loadLessonsMock,
            removeLessons: removeLessonsMock,
            setLessonsPagination: setLessonsPaginationMock,
            setSelectedLesson: setSelectedLessonMock,
            finishOrStartLesson: jest.fn(),
        });

        const { getByTestId } = render(
            <MenuProvider>
                <LessonsList />
            </MenuProvider>
        );

        /* Get empty message of list and check if is text pass for props */
        const emptyMsgText = getByTestId('info-text-text');
        expect(emptyMsgText.props.children).toBe(emptyMessageTest);
    });

    it('should render loading when isCoursesLoading is true', () => {

        /* Mock data of useCourses */
        (useCourses as jest.Mock).mockReturnValue({
            state: {
                ...courseSelectedState,
                isLessonsLoading: true
            },
            deleteLesson: deleteLessonMock,
            loadLessons: loadLessonsMock,
            removeLessons: removeLessonsMock,
            setLessonsPagination: setLessonsPaginationMock,
            setSelectedLesson: setSelectedLessonMock,
            finishOrStartLesson: jest.fn(),
        });

        const { getByTestId } = render(
            <MenuProvider>
                <LessonsList />
            </MenuProvider>
        );

        /* Get loader and check if exists in component */
        const loader = getByTestId('loader');
        expect(loader).toBeTruthy();
    });

    it('should search when searchInput is submit', () => {

        /* Get search input text, type search and submit */
        const searchInput = screen.getByTestId('search-input-text-input');
        fireEvent(searchInput, 'onChangeText', 'Test search');
        fireEvent(searchInput, 'onSubmitEditing');

        /**
         * Check if setLessonsPagination, removeLessons and loadLessons is called
         * one time with respective args
         */
        expect(setLessonsPaginationMock).toHaveBeenCalledTimes(1);
        expect(setLessonsPaginationMock).toHaveBeenCalledWith({ from: 0, to: 9 });
        expect(removeLessonsMock).toHaveBeenCalledTimes(1);
        expect(loadLessonsMock).toHaveBeenCalledTimes(1);
        expect(loadLessonsMock).toHaveBeenCalledWith({
            search: 'Test search', refresh: true
        });
    });
});