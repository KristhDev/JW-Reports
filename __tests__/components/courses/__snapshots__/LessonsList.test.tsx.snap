// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <LessonsList /> component should to match snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      [
        {
          "flex": 1,
        },
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    <RCTScrollView
      ListEmptyComponent={
        <ListEmptyComponent
          msg="No haz agregado clases a este curso."
          showMsg={false}
        />
      }
      ListFooterComponent={
        <ListFooterComponent
          marginTopPlus={false}
          showLoader={false}
        />
      }
      ListHeaderComponent={
        <React.Fragment>
          <Title
            containerStyle={
              {
                "marginVertical": 8,
              }
            }
            text="Clases del curso con Fred Cremin"
            textStyle={
              {
                "fontSize": 24,
              }
            }
          />
          <SearchInput
            onClean={[Function]}
            onSearch={[Function]}
            refreshing={false}
            searchTerm=""
          />
        </React.Fragment>
      }
      ListHeaderComponentStyle={
        {
          "alignSelf": "flex-start",
        }
      }
      contentContainerStyle={
        {
          "alignItems": "center",
          "flexGrow": 1,
          "padding": 24,
          "paddingBottom": 100,
        }
      }
      data={
        [
          {
            "courseId": "8ea3baf8-6b47-4222-8ed1-c62d5755b9a5",
            "createdAt": "2021-01-01T00:00:00.000Z",
            "description": "Nemo et et non nesciunt ipsum sed rerum vero. Nostrum quae officiis qui nihil dignissimos consectetur aperiam. Vel harum ex. Ipsum ratione necessitatibus eum sint vero enim et consequatur. Ab autem rerum numquam ut. Laboriosam nihil est exercitationem unde aut reiciendis odit.",
            "done": false,
            "id": "0fb579e4-de87-4336-9375-76611c889f76",
            "nextLesson": "2021-01-01T00:00:00.000Z",
            "updatedAt": "2021-01-01T00:00:00.000Z",
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onEndReached={[Function]}
      onEndReachedThreshold={0.5}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onRefresh={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      overScrollMode="never"
      refreshControl={
        <RefreshControlMock
          onRefresh={[Function]}
          refreshing={false}
        />
      }
      refreshing={false}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <RCTRefreshControl />
      <View>
        <View
          collapsable={false}
          onLayout={[Function]}
          style={
            {
              "alignSelf": "flex-start",
            }
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "flex-start",
                  "marginBottom": 40,
                  "width": "100%",
                },
                {
                  "marginVertical": 8,
                },
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#FFFFFF",
                    "fontSize": 38,
                    "fontWeight": "bold",
                  },
                  {
                    "fontSize": 24,
                  },
                ]
              }
              testID="title-text"
            >
              Clases del curso con Fred Cremin
            </Text>
          </View>
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "marginBottom": 24,
                "paddingHorizontal": 8,
                "width": "100%",
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": "#BFBFBF",
                    "fontSize": 25,
                  },
                  undefined,
                  {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
            >
              
            </Text>
            <View
              style={
                {
                  "alignItems": "center",
                  "borderBottomWidth": 1,
                  "borderColor": "#BFBFBF",
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
              testID="search-input-text-input-container"
            >
              <TextInput
                autoCorrect={false}
                cursorColor="#C0A7E1"
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                onSubmitEditing={[Function]}
                placeholder="Buscar"
                placeholderTextColor="#BFBFBF"
                returnKeyType="search"
                style={
                  {
                    "color": "#FFFFFF",
                    "flex": 1,
                    "fontSize": 17,
                    "paddingLeft": 10,
                  }
                }
                testID="search-input-text-input"
                value=""
              />
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": true,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                centered={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "overflow": "hidden",
                    },
                    {
                      "alignItems": "center",
                      "borderRadius": 999,
                      "height": 40,
                      "justifyContent": "center",
                      "width": 40,
                    },
                  ]
                }
                testID="search-input-clear-btn"
              >
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    [
                      {
                        "color": "transparent",
                        "fontSize": 30,
                      },
                      undefined,
                      {
                        "fontFamily": "Ionicons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      {},
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "overflow": "hidden",
                },
                {
                  "marginVertical": 8,
                  "width": 734,
                },
              ]
            }
            testID="lesson-card-touchable"
          >
            <View
              style={
                {
                  "backgroundColor": "#292929",
                  "padding": 15,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#BFBFBF",
                    "fontSize": 14,
                    "marginBottom": 15,
                  }
                }
                testID="lesson-card-status-text"
              >
                Clase para el 31 de December del 2020
              </Text>
              <Text
                style={
                  {
                    "color": "#FFFFFF",
                    "fontSize": 16,
                  }
                }
                testID="lesson-card-description-text"
              >
                Nemo et et non nesciunt ipsum sed rerum vero. Nostrum quae officiis qui nihil dignissimos consectetur aperiam. Vel harum ex. Ipsum ratione necessitatibus eum sint vero enim et consequatur. Ab autem re...
              </Text>
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": false,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "overflow": "hidden",
                    },
                    [
                      {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "borderRadius": 50,
                        "height": 48,
                        "justifyContent": "center",
                        "margin": 5,
                        "width": 48,
                      },
                      {
                        "height": 35,
                        "position": "absolute",
                        "right": -2.5,
                        "top": 5,
                        "width": 35,
                      },
                    ],
                  ]
                }
                testID="fab-touchable"
              >
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    [
                      {
                        "color": "#C0A7E1",
                        "fontSize": 21,
                      },
                      undefined,
                      {
                        "fontFamily": "Ionicons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      {},
                    ]
                  }
                >
                  
                </Text>
              </View>
              <View
                style={
                  {
                    "position": "absolute",
                    "right": 20,
                    "top": 30,
                  }
                }
              >
                <View
                  collapsable={false}
                >
                  <View
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                  >
                    <View
                      ctx={
                        {
                          "menuActions": {
                            "_getOpenedMenu": [Function],
                            "_notify": [Function],
                            "closeMenu": [Function],
                            "isMenuOpen": [Function],
                            "openMenu": [Function],
                            "toggleMenu": [Function],
                          },
                          "menuRegistry": {
                            "getAll": [Function],
                            "getMenu": [Function],
                            "setOptionsCustomStyles": [Function],
                            "subscribe": [Function],
                            "unsubscribe": [Function],
                            "updateLayoutInfo": [Function],
                          },
                        }
                      }
                      style={
                        [
                          undefined,
                          undefined,
                        ]
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
        />
      </View>
    </RCTScrollView>
    <Modal
      animationType="fade"
      hardwareAccelerated={false}
      statusBarTranslucent={true}
      transparent={true}
      visible={false}
    />
    <Modal
      animationType="fade"
      hardwareAccelerated={false}
      statusBarTranslucent={true}
      transparent={true}
      visible={false}
    />
  </View>
  <RCTSafeAreaView
    pointerEvents="box-none"
    style={
      {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      collapsable={false}
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        {
          "flex": 1,
        }
      }
    />
  </RCTSafeAreaView>
</View>
`;
